statesync/syncer.go

```diff
@@ -161,6 +161,7 @@ func (s *syncer) SyncAny(discoveryTime time.Duration) (sm.State, *types.Commit,
 			if discoveryTime == 0 {
 				return sm.State{}, nil, errNoSnapshots
 			}
+			retryHook()
```

Comment:
I think resending the snapshot request makes sense because there's a chance there will be more peers that the node is connected with that will have a suitable snapshot. I wonder if we need to put anything in place to avoid setting a discovery time that is super low

Reply:
What do you think is a reasonable minimum?

Reply:
So default is 15 seconds. I think the e2e tests are 5s so I would also say 5s is enough (maybe even less)

Reply:
Note that I think we should still allow for the case where discovery time is 0

Reply:
sounds good, I did a thing